core/src/cluster_info.rs

```diff
@@ -307,10 +308,75 @@ impl ClusterInfo {
         )
     }
 
-    pub fn push_epoch_slots(&mut self, id: Pubkey, root: Slot, min: Slot, slots: BTreeSet<Slot>) {
+    pub fn compress_incomplete_slots(incomplete_slots: &BTreeSet<Slot>) -> (Slot, Vec<u8>) {
+        if !incomplete_slots.is_empty() {
+            let first_slot = incomplete_slots
+                .iter()
+                .next()
+                .expect("expected to find at least one slot");
+            let last_slot = incomplete_slots
+                .iter()
+                .last()
```

Comment:
Because this iterator implements `DoubleEndedIterator `, `next_back()` is a more efficient way to get the last item without iterating over the entire set

Reply:
LMK when you are done with the review. I'll update the PR with the required changes.